includes/blocks/class-sensei-blocks.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * File containing the class Sensei_Blocks.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Blocks
+ */
+class Sensei_Blocks {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Sensei_Blocks constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Initialize the class.
+	 *
+	 * @param Sensei_Main $sensei_main_instance Sensei main class instance.
+	 */
+	public function init( $sensei_main_instance ) {
+		// Skip if Gutenberg is not available.
+		if ( ! function_exists( 'register_block_type' ) ) {
+			return;
+		}
+
+		add_filter( 'block_categories', [ $this, 'sensei_block_categories' ], 10, 2 );
+
+		// Init blocks.
+		if ( $sensei_main_instance->feature_flags->is_enabled( 'course_outline' ) ) {
+			new Sensei_Course_Outline_Block();
```

Comment:
For now, a block class looks very simple. If in the future, with more blocks, we identify some pattern that we want always use, could be a good idea creating an interface class for them.